openupgradelib/cleanup.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright (C) 2011-2013 Therp BV (<http://therp.nl>)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import logging
+from .openupgrade import column_exists
+# from .openupgrade import version_info
+
+logger = logging.getLogger('OpenUpgradeCleanup')
+logger.setLevel(logging.DEBUG)
+
+###########################################
+#
+# NOTE: Please try to homologate this library with
+# database_cleanup module as you contribute
+# You can find it here:
+# https://github.com/OCA/server-tools/tree/9.0/database_cleanup
+#
+# Feel yourself encouraged to contact the author of a
+# similar method in databse_cleanup (git blame).
+#
+###########################################
+
+
+__all__ = [
+    'drop_m2m_table',
+    'drop_columns',
+]
+
+
+def drop_m2m_table(env, table_spec):
+    """
+    Drop a many2many relation table properly.
+    You will typically want to use it in post-migration scripts after you \
+    have migrated the values of your many2many fields.
+
+    :param cr: The database cursor
+    :param table_spec: list of strings ['table one', 'table two']
+
+    .. versionadded:: 9.0
```

Comment:
This is meaningless now that openupgradelib is shared between all versions. But you will want to make a remark that this method will not work before 7.0 as m2m relations were not registered in this way back then.